tests/chainer_tests/functions_tests/normalization_tests/test_l2_normalization.py

```diff
@@ -26,14 +45,32 @@
         {'shape': (), 'axis': ()},
     ],
     [
-        {'eps': 1e-5},
-        {'eps': 1e-1},
+        {'eps': 1e-5, 'nonzeros': None},
+        {'eps': 1e-1, 'nonzeros': None},
+        {'eps': 1e-1, 'nonzeros': 0, 'truezero': True},
+        {'eps': 1e-1, 'nonzeros': 0, 'truezero': False},
+        {'eps': 1e-1, 'nonzeros': 2, 'truezero': True},
+        {'eps': 1e-1, 'nonzeros': 2, 'truezero': False},
     ],
 ]))
 class TestL2Normalization(unittest.TestCase):
 
     def setUp(self):
         self.x = numpy.random.uniform(-1, 1, self.shape).astype(numpy.float32)
+        if self.nonzeros is not None:
```

Comment:
Please add comments for the logic.

Reply:
Please write a brief comment here.
(Maybe single line is enough here, if the detail is covered in the comments of respective arguments.)